modules/collection/environment/warning.nix

```
@@ -0,0 +1,30 @@
+{
+  lib,
+  config,
+  ...
+}: let
+  inherit (lib.options) mkEnableOption;
+  inherit (lib.modules) mkIf;
+  inherit (lib.lists) optionals;
+  inherit (builtins) all;
+
+  # A list checking all modules that load variables. Enabled modules
+  # will evaluate to true, disabled will evaluate to false. Results
+  # in a list of bools, allowing us to
+  varsLoaders = [
+    config.rum.programs.zsh.enable
+    config.rum.programs.fish.enable
+    config.rum.programs.hyprland.enable
+  ];
+
+  cfg = config.rum.environment;
+in {
+  options.rum.environment.hideWarning = mkEnableOption "a warning for env vars not being implemented";
+
+  config = mkIf (!cfg.hideWarning) {
+    # If all modules are disabled, then Hjem Rum is not loading variables
+    warnings = optionals ((all (module: !module) varsLoaders) && (config.environment.sessionVariables != {})) [
+      "environment.sessionVariables exist but are not being loaded by any Hjem Rum modules. This is not done natively by Hjem, and unless you are loading your sessionVariables manually, may cause problems. If this is unintended, enable a module that loads sessionVariables, such as a shell. If you would like to disable this warning, enable rum.environment.hideWarning."
```

```suggestion
      "environment.sessionVariables exist but are not being loaded by any Hjem Rum modules. 
      This is not done natively by Hjem, and unless you are loading your sessionVariables manually, 
      it may cause problems. If this is unintended, enable a module that loads sessionVariables, such as a shell. 
      If you would like to disable this warning, enable rum.environment.hideWarning."
```

I would split this onto multiple lines. Also you forgot "it" before "may cause problems"

---

I would just split it in two then:

```nix
''
environment.sessionVariables exist but are not being loaded by any Hjem Rum modules. Please see the README for more information. 
If you would like to disable this warning, enable rum.environment.hideWarning.
'''
```

Just so the warning disable stuff is on its own line